consensus/spos/scheduledProcessor.go

```diff
@@ -149,6 +159,44 @@ func (sp *scheduledProcessorWrapper) StartScheduledProcessing(header data.Header
 	}()
 }
 
+// IsInterfaceNil returns true if there is no value under the interface
+func (sp *scheduledProcessorWrapper) IsInterfaceNil() bool {
+	return sp == nil
+}
+
+func (sp *scheduledProcessorWrapper) emptyStopChannel() {
```

Comment:
this resembled with the core.EmptyChannel function. could have used that by changing stopExecution to be a buffered chan of type bool.